### DISCUSSION TITLE
Building dynamic attachment templates in block themes

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#192] [Topic Ideas]
Block themes support `attachment.html` templates, but WordPress is not very "smart" about handling the output of media for attachment views. At the moment, there are no dynamic Image, Audio, and Video blocks that automatically pick up the current attachment media file on the front end.

Yes, attachments views are now disabled by default for new WP installs (as of 6.4?), but there are still many use cases where attachment templates are needed.

Typically, I don't like to propose topics for the Dev Blog that are outside the standards of theme dev, but this is one of those situations that requires a bit of hacking to make things work as well as they did in classic. That's unfortunate, but this is a missing piece of block theme dev with no instructions/docs (AFAIK) on the best way to handle this.

I'm proposing a tutorial on how to filter the Post Content block for attachment views to insert a sort of dynamic template partial/pattern that allows theme authors to control the output of attachment pages.

I have a working example in my own theme that I could break down for this tutorial:

- Filter: https://github.com/x3p0-dev/x3p0-ideas/blob/da7b5abc87dd05312b58d1ac49cf699ac9ef8800/src/Blocks.php#L31-L115
- Partials: https://github.com/x3p0-dev/x3p0-ideas/tree/da7b5abc87dd05312b58d1ac49cf699ac9ef8800/public/partials

**Comment by bph** [comment]
Discussion locked, conversation moved to the issue

- #199
